### ISSUE TITLE
multiple Tugboat demo environments showing 504 Gateway Time-out error

### ISSUE THREAD

**Issue by xiongjaneg** [department-of-veterans-affairs/va.gov-cms#8094]
## Describe the defect
At 1:37 p.m. ET today, on the [Claro demo environment](https://clarodemo-x1iumxudopspquvg1n2h3ylucye7xlhp.demo.cms.va.gov/), the page suddenly displayed a 504 Gateway Time-out error. Trying the [training demo environment](https://cms-training-1hegto5jxzzxaqtrkgv1fqcvatavb4rr.demo.cms.va.gov/) produced the same result.

## To Reproduce
1. Go to https://cms-training-1hegto5jxzzxaqtrkgv1fqcvatavb4rr.demo.cms.va.gov/
2. See error 504 Gateway Time-out error

## Expected behavior
The demo environment login page should be displayed.

## Additional context
Add any other context about the problem here. Reach out to the Product Managers to determine if it should be escalated as critical (prevents users from accomplishing their work with no known workaround and needs to be addressed within 2 business days).

 
## Desktop (please complete the following information if relevant, or delete)
 - OS: [e.g. iOS] MacOS Catalina
 - Browser [e.g. chrome, safari] Chrome
 - Version [e.g. 22] Chrome Version 98.0.4758.102 (Official Build) (x86_64)

## Labels
(You can delete this section once it's complete)
- [x] Issue type (red) (defaults to "Defect")
- [ ] CMS subsystem (green)
- [ ] CMS practice area (blue)
- [x] CMS workstream (orange) (not needed for bug tickets)
- [ ] CMS-supported product (black)

### CMS Team
Please check the team(s) that will do this work. 

- [ ] `CMS Program`
- [x] `Platform CMS Team`
- [ ] `Sitewide CMS Team ` (leave Sitewide unchecked and check the specific team instead)
  - [ ] `⭐️ Content ops`
  - [ ] `⭐️ CMS experience`
  - [ ] `⭐️ Offices`
  - [ ] `⭐️ Product support`
  - [ ] `⭐️ User support`

**Comment by xiongjaneg** [comment]
[See Slack for initial post](https://dsva.slack.com/archives/CT4GZBM8F/p1645209425460079)

**Comment by xiongjaneg** [comment]
Posted notification on [virtual agent](https://dsva.slack.com/archives/C01KTS3F493/p1645214175922269) and [content release engineering ](https://dsva.slack.com/archives/C01512KN35G/p1645214198506209)channels

**Comment by ElijahLynn** [comment]
@ndouglas and I worked on this today we resolved this issue around 3:45pm ET and figured out the root cause. 

A base preview refresh/rebuild was triggered somewhere around 12pm ET (maybe earlier) and the checkbox for "Rebuild previews built from this one when finished" was checked. This was fine in and of itself as Tugboat staggers the builds to happen one or two at a time. The challenge happened when as they finished building they started running tests, which over the months the test runs have gotten longer and longer and longer (they run two full content builds per test suite run) which ultimately had 30+ test suite runs (1 for each PR) happening concurrently, and with 1.5GB of ram minimum per enviornment when tests are not running, and they likely spiked to 2-4GB each and we had a memory exhaustion scenario as the server has 128GB of ram.
This was a good idea if we needed certain code to go out and nobody did anything wrong, just to clarify. But we are in a scenario now where because of the amount of open PRs, this is a death blow to Tugboat. So, until this is resolved, we cannot check that box on a base preview rebuild. We are going to noodle on the best solution for this (e.g. upping resources, pruning environments, adding swap + [nohang](https://github.com/hakavlad/nohang), fixing test runs etc). 

![image](https://user-images.githubusercontent.com/1504756/154778759-31e577ee-cab3-4dfd-aee8-d7ce530e22cf.png)

I still owe a better write up of what happened but this is better than nothing for right now.
